### DISCUSSION TITLE
Example EthernetWebServer/examples/WiFi_Ethernet_Complex_ESP32

### DISCUSSION THREAD

**Discussion by simogaspa84** [khoih-prog/EthernetWebServer#51] [General]
Hi ... 

Thanks for sharing this library.. 
Have you ever tried this example on this enc28j60 chip ? 
I am going to use it with ESP32 with Arduino framework.. 

Thanks a lot

**Comment by khoih-prog** [comment]
I believe it's OK to use `ENC28J60` + `EthernetENC` using this [**EthernetWebServer** library](https://github.com/khoih-prog/EthernetWebServer) for ESP32.

The connections are as follows:

```
MOSI:23
MISO:19
SCK:18
CS: 22
```

Remember to use [**7. For fixing ESP32 compile error**](https://github.com/khoih-prog/EthernetWebServer#7-for-fixing-esp32-compile-error)

If you still have issue, I'm afraid I won't have time to help you any further.

---

Even though it's possible to use ENC28J60, **it's not advisable** in the long run. It's the **best** to use ESP32 with **LAN8720**, such as `WT32_ETH01` boards (ESP32 with built-in LAN8720), then **second best** is ESP32 + W5500
